replica/server.go

```diff
@@ -354,10 +357,37 @@ func (s *Server) Restore(url, name string) (err error) {
 		return err
 	}
 
-	if err := backupstore.RestoreDeltaBlockBackup(backupURL, toFile); err != nil {
+	restoreObj := NewRestore(toFile)
+
+	config := &backupstore.DeltaRestoreConfig{
+		BackupURL:    backupURL,
+		DeltaOps:     restoreObj,
+		SnapshotName: toFile,
+	}
+
+	if err := backupstore.RestoreDeltaBlockBackup(config); err != nil {
+		logrus.Errorf("failed to initiate Restore backup for url[%v]: %v", backupURL, err)
 		return err
 	}
 
+	for j := 0; j < RetryAttempts; j++ {
```

Comment:
There are two issues with this approach.

1. This operation should be an async process. The caller shouldn't be blocked once the restoration started.
2. The polling mechanism cannot be time-based. It's the biggest issue of the previous approach. Because we cannot know how long a backup or restoration will take, so we cannot say for sure it would be done in 1800 seconds. With progress, what we can do instead, is there should be updated in e.g. 5 minutes. It shouldn't be just percentage change because it's 100 percent at most. Seems we would need a timestamp on when is last time status was updated, as long as it has been updated at least once in e.g. 5 minutes, we know we're making progress.